src/puppetlabs/puppetdb/command/dlo.clj

```diff
@@ -1,210 +1,173 @@
-  [subdir threshold]
-  {:pre [(or (string? subdir)
-             (instance? java.io.File subdir))]}
-  (if-let [archive-time (last-archived subdir)]
-    (after? archive-time (ago threshold))))
-
-(defn- compress-subdir!
-  "Compresses the specified `files` in the particular command-specific
-  subdirectory `subdir`, replacing them with a timestamped .tgz file."
-  [subdir files]
-  {:pre [(or (string? subdir)
-             (instance? java.io.File subdir))]}
-  (create-metrics-for-subdir! subdir)
-  (let [target-file (file subdir (str (kitchensink/timestamp) ".tgz"))
-        temp (str target-file ".partial")]
-    (try
-      (time! (subdir-metric subdir :compression)
-             (apply archive/tar temp "UTF-8" (map (juxt fs/base-name slurp) files))
-             (fs/rename temp target-file)
-             (doseq [filename files]
-               (fs/delete filename)))
-      (catch Throwable e
-        (mark! (global-metric :compression-failures))
-        (mark! (subdir-metric subdir :compression-failures))
-        (throw e))
-      (finally
-        (fs/delete temp)))))
-
-(defn compress!
-  "Compresses all DLO subdirectories which have messages that have been in the
-  directory for longer than `threshold`. This will produce a timestamped .tgz
-  file in each subdirectory containing old messages."
-  [dir threshold]
-  {:pre [(or (string? dir)
-             (instance? java.io.File dir))
-         (period? threshold)]}
-  (create-metrics-for-dlo! dir)
-  (when-let [subdir-files (seq (for [subdir (remove #(already-archived? % threshold) (subdirectories dir))
-                                     :let [files (compressible-files subdir threshold)]
-                                     :when (seq files)]
-                                 [subdir files]))]
-    (time! (global-metric :compression)
-           (doseq [[subdir files] subdir-files]
-             (compress-subdir! subdir files)))))
+  (:require
+   [clojure.java.io :as io]
+   [clojure.string :as str]
+   [puppetlabs.i18n.core :refer [trs]]
+   [puppetlabs.kitchensink.core :refer [timestamp]]
+   [puppetlabs.puppetdb.nio :refer [copts copt-atomic copt-replace]]
+   [puppetlabs.puppetdb.queue :refer [cmdref->entry parse-metadata]]
+   [metrics.gauges :refer [gauge-fn]]
+   [stockpile :as stock])
+  (:import
+   [java.nio.file AtomicMoveNotSupportedException
+    FileAlreadyExistsException Files LinkOption]
+   [java.nio.file.attribute FileAttribute]))
+
+(defn- parse-cmd-filename [s]
+  (let [rx #"([0-9+])-(.*)"]
+    (when-let [[_ id qmeta] (re-matches rx s)]
+      (parse-metadata qmeta))))
+
+(defn- update-cmd-stats
+  "Updates stats to reflect the receipt of the named command."
+  [stats command size]
+  (let [addnp (fn [v n] (+ (or v 0) n))]
+    (-> stats
+        (update-in ["all" :count] inc)
+        (update-in ["all" :size] addnp size)
+        (update-in [command :count] addnp 1)
+        (update-in [command :size] addnp size))))
+
+(defn- gather-stats [dlo]
```

Comment:
Elsewhere in the code, `dlo` is the atom containing the metrics, here `dlo` means the dlo directory. That tripped me up a bit.